R/plotCCA.R

```diff
@@ -0,0 +1,504 @@
+#'  plotRDA(tse, "RDA",
+#'          colour_by = "ClinicalStatus,
+#'          repel_text = FALSE)
+#'  
+#'  
+NULL
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setGeneric("plotCCA", signature = c("object"),
+           function(object, dimred, ...) standardGeneric("plotCCA"))
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setMethod("plotCCA", signature = c(object = "SingleCellExperiment"),
+    function(object, dimred, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, dimred, ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotRDA
+#' @export
+setMethod("plotCCA", signature = c(object = "matrix"),
+    function(object, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Construct TreeSE from rda/cca object
+        object <- .rda2tse(object)
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, "RDA", ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setGeneric("plotRDA", signature = c("object"),
+    function(object, dimred, ...) standardGeneric("plotRDA"))
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setMethod("plotRDA", signature = c(object = "SingleCellExperiment"),
+    function(object, dimred, ...){
+        ###################### Input check #######################
+        
+        ###################### Input check end ####################
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, dimred, ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+#' @rdname plotRDA
+#' @aliases plotCCA
+#' @export
+setMethod("plotRDA", signature = c(object = "matrix"),
+    function(object, ...){
+        ###################### Input check #######################
+            
+        ###################### Input check end ####################
+        # Construct TreeSE from rda/cca object
+        object <- .rda2tse(object)
+        # Get data for plotting
+        plot_data <- .incorporate_rda_vis(object, "RDA", ...)
+        # Create a plot
+        plot <- .rda_plotter(plot_data, ...)
+        return(plot)
+    }
+)
+
+
+################## HELP FUNCTIONS ##########################
+
+# Construct TreeSE from rda/cca object to pass it to downstream functions
+.rda2tse <- function(object) {
+    # If rda/cca object, convert it to TreeSE
+    if (is.matrix(object)) {
+        # Remove significance attribute if present
+        attr(rda, "significance") <- NULL
```

Comment:
How about including this? So just keep it and put the matrix as it is to reducedDim? If the input is matrix, it should be from calculateRDA (if it is not, then .incorporate_rda_vis should handle the issue

Reply:
Also there is no rda variable